history_continue.c

```diff
@@ -0,0 +1,42 @@
+#include "history.h"
+#include <stdlib.h>
+#include "libft.h"
+#include "ft_printf.h"
+
+t_history	*history_create(void)
+{
+	const size_t	vec_size = 100;
+	t_history		*hist;
+
+	hist = (t_history *)malloc(sizeof(t_history));
+	if (!hist || !vector(&hist->vec, V_CREATE, vec_size, NULL))
+	{
+		free(hist);
+		return (NULL);
+	}
+	hist->size = 0;
+	hist->current_index = 0;
+	return (hist);
+}
+
+char	*history_get_last(t_history *hist)
+{
+	if (hist->size == 0)
+		return (NULL);
```

Comment:
```suggestion
```

`vector()` protects against out of boundary problems, so you can drop this check